hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -868,7 +876,31 @@ def _deperecated_terminate_status(self, input_json):
                 
                 if machines_to_terminate:
                     logger.warning("Re-attempting termination of nodes %s", machines_to_terminate)
-                    self.cluster.terminate(machines_to_terminate)
+                    try:
+                       self.cluster.terminate(machines_to_terminate)
+                    except requests.exceptions.ConnectionError:
+                       logger.warning("Could not terminate machines %s due to connection exception, sending status as running", machines_to_terminate)
+                       request_status = RequestStates.running
+                       response["status"] = request_status
+                       response_machines = []
+
+                       for termination_id in termination_ids:
+                            # if we don't know the termination_id then we report an empty list of machines
+                            request = {"requestId": termination_id,
+                                       "machines": response_machines}
+                            # report machines are in deleting state so HF remembers the request 
+                            if termination_id in terminate_requests:
+                                 termination_request = terminate_requests.get(termination_id)
+                                 machines = termination_request.get("machines", {})
+                                 if machines:
+                                    for machine_id, hostname in machines.items():
+                                        response_machines.append({"name": hostname,
+                                                    "status": MachineStates.deleting,
+                                                    "result": MachineResults.executing,
+                                                    "machineId": machine_id})  
+                            response["requests"].append(request) 
+                       return self.json_writer(response)
+                       
```

Comment:
Probably we can remove the outer try: except: now

We should catch ALL exceptions here.